app/controllers/registrations_controller.rb

```diff
@@ -26,6 +26,26 @@ def disable_two_factor
     end
   end
 
+  def tour
+    @user = User.find(params[:id])
+    if @user.update(params.require(:user).permit(:tour, :tour_steps))
+      render json: @user, status: :ok
+    else
+      render json: @user.errors, status: :unprocessable_entity
+    end
+  end
+
+  def reset_tour
+    @user = User.find(params[:id])
+    if @user.update(tour: true, tour_steps: WelcomeTour::STEPS.to_json)
+      flash[:notice] = t('reset_tour_success')
```

Comment:
You can make it better with just `redirect_to :back, flash: { notice: t('reset_tour_success') }`